### PR TITLE
refactor with configuration and service modules

### PR DESCRIPTION
**Pull request by erikdstock**
This refactor
- moves references to the ENV into a single CONFIG openstruct constant accessible via `Artsy::EventService.config`
- Moves RabbitMQ connection logistics into its own module and namespaces publishing logic
- Lays the groundwork for a queue consumer functionality, the logic for which already exists in artsy/pulse#68

It should not break any existing functionality (but please help me verify this?)- minor version bump to `0.1.4` but the consumer functionality following this will be `0.2`.

### CODE REVIEW

**Comment by bhoggard** [comment]
👍 I don't have any comments.

File: lib/artsy-eventservice/artsy/event_service/rabbitmq_connection.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+module Artsy
+  module EventService
+    module RabbitMQConnection
+      # get a new RabbitMQ Client
+      def create_conn
+        Bunny.new(rabbitmq_url, **bunny_params)
+      end
+
+      # Connect, do something and close the connection
+      def connect_to_rabbit
```
Comment (izakp) [inline lib/artsy-eventservice/artsy/event_service/rabbitmq_connection.rb]: In a consumer, we will want to keep connections open, while in the case of a publisher, although it is okay to create and destroy a conn object, it does incur some overhead.  I would recommend opening a connection per thread, and leaving it open, allowing bunny's `automatic_recovery` to handle keeping the connection open.  If we want to have different behavior in consumers and producers, I would move the `connect_to_rabbit` method to the Producer / Consumer classes.  See: http://rubybunny.info/articles/error_handling.html for connection error handling and  http://rubybunny.info/articles/concurrency.html for thread-safety in bunny

File: spec/artsy-eventstream/artsy/publisher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    end
+    it 'fails when event.verb is nil' do
+      allow(event).to receive(:verb).and_return(nil)
+      expect { Artsy::EventService::Publisher.publish(topic: 'test', event: event) }.to raise_error 'Event missing topic or verb.'
+    end
+    it 'calls publish on the exchange with proper data' do
+      conn = double
+      channel = double
+      exchange = double
+      allow(Bunny).to receive(:new).and_return(conn)
```
Comment (izakp) [inline spec/artsy-eventstream/artsy/publisher_spec.rb]: Nice!

**Comment by izakp** [comment]
Nicely done @erikdstock ! My only comment is about re-using connections, but we can address that in a further version when we implement consumers.

File: lib/artsy-eventservice/artsy/event_service/rabbitmq_connection.rb (hunk 1 above)
Comment (izakp) [inline lib/artsy-eventservice/artsy/event_service/rabbitmq_connection.rb]: Yea so ideally, we want to open one connection per process, and then 1 channel per thread.  Further reading: 
- http://rubybunny.info/articles/connecting.html
- http://stackoverflow.com/questions/10407760/is-there-a-performance-difference-between-pooling-connections-or-channels-in-rab

I wonder how we could enforce this though, given that this is a shared library, imported into different applications (i.e. some Rails projects using Unicorn, where the connection should be set up in the `:after_fork` block, and some projects using Puma, which is a threaded process worker model) - thoughts @ashkan18 ?

File: lib/artsy-eventservice/artsy/event_service/publisher.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    class Publisher
+      include RabbitMQConnection
+
+      def self.publish(topic:, event:)
+        new.post_event(topic: topic, event: event)
+      end
+
+      def post_event(topic:, event:)
+        raise 'Event missing topic or verb.' if event.verb.to_s.empty? || topic.to_s.empty?
+        connect_to_rabbit do |conn|
```
Comment (izakp) [inline lib/artsy-eventservice/artsy/event_service/publisher.rb]: So if we create the connection as `@conn` when the EventService class is instantiated, then create a method `with_channel` instead:
```
def with_channel
  channel = @conn.create_channel
  yield channel
  channel.close
end
```

And use it like:
```
with_channel do |channel|
  exchange = channel.topic(topic, durable: true)
  ...
end
```

**Pull request merged**